src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -70,12 +70,15 @@ protected override SyntaxNode InsertNamespaceImport(SyntaxNode root, SyntaxGener
                     : UsingsAndExternAliasesDirectiveComparer.NormalInstance;
 
             // find insertion point
+            int existingIndex = 0;
             foreach (var existingImport in gen.GetNamespaceImports(root))
             {
                 if (comparer.Compare(import, existingImport) < 0)
                 {
-                    return gen.InsertNodesBefore(root, existingImport, new[] { import });
```

Comment:
Whatever fix you do do make the banner trivia get preserved in the InsertNamespaceImports API should also be true when InsertNodesBefore is called, or ReplaceNode, etc.